### PR TITLE
Issue 7 feat/app api setup

### PR DESCRIPTION
**Pull request by T0ffik**

### CODE REVIEW

File: src/Utils/ToastNotify.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import 'react-toastify/dist/ReactToastify.css';
+
+import { toast } from 'react-toastify';
+
+export const toastNotify = (status: number, text?: string) => {
+    switch (status) {
+        case 400:
+            return toast.error(
+                text ? text : 'Bad request, try something else!',
```
Comment (Veryyapeee) [inline src/Utils/ToastNotify.ts]: Just
`text || ...`

File: src/Utils/ToastNotify.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            return toast.error(text ? text : 'Unauthorized');
+        case 404:
+            return toast.error(text ? text : 'Not Found');
+        case 500:
+            return toast.info(
+                text ? text : 'Internal server error, try again later'
+            );
+        case 200:
+            return toast.success(text ? text : 'Success!!', {
+                position: 'bottom-right',
```
Comment (Veryyapeee) [inline src/Utils/ToastNotify.ts]: move shared styles to separate variable

File: src/Utils/ToastNotify.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                    closeOnClick: true,
+                    pauseOnHover: false,
+                    draggable: false,
+                    progress: undefined
+                }
+            );
+        case 401:
+            return toast.error(text ? text : 'Unauthorized');
+        case 404:
+            return toast.error(text ? text : 'Not Found');
```
Comment (Veryyapeee) [inline src/Utils/ToastNotify.ts]: same as up etc.

File: src/Utils/types.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+export type Login = {
+    username: string;
+    password: string;
+};
+
+export type MappedResponse<T> = {
+    data: boolean | T | undefined;
```
Comment (Veryyapeee) [inline src/Utils/types.ts]: `data?: boolean | T`

Instead of **| undefined**

File: src/hooks/useApi.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    args: G,
+    tag: string
+) {
+    const { data, isError, isLoading } = useQuery(tag, async () => {
+        try {
+            const data = await callback(args);
+            return data.data;
+        } catch (err: any) {
+            const isError = true;
+            toastNotify(err.response.status);
```
Comment (Veryyapeee) [inline src/hooks/useApi.tsx]: We should have flag for enabling/disabling toast's, we don't want to display toast every time

File: src/hooks/useApi.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+export function useRequest<T, G>(
+    callback: (config: G) => Promise<AxiosResponse<T, G>>,
+    args: G,
+    tag: string
+) {
+    const { data, isError, isLoading } = useQuery(tag, async () => {
+        try {
+            const data = await callback(args);
+            return data.data;
```
Comment (Veryyapeee) [inline src/hooks/useApi.tsx:19]: also same flag here, you added handling for success toast, but now you are not displaying it anywhere

File: src/Axios/Axios.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import axios, { AxiosResponse } from 'axios';
+
+import { AuthService } from '../Api/client/AuthService';
+
+export const instance = axios.create({
+    baseURL: process.env.REACT_APP_API_URL,
```
Comment (Veryyapeee) [inline src/Axios/Axios.ts:6]: add this to .env.example

File: src/Utils/types.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
+export type Login = {
```
Comment (Veryyapeee) [inline src/Utils/types.ts]: I think that creating huge file for every type is kinda bad

I suggest to create file /types/requestTypes (smth like that, come up with better name pls)

File: src/hooks/useApi.tsx [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+        } catch (err: any) {
```
Comment (Veryyapeee) [inline src/hooks/useApi.tsx]: we don't do that here

File: src/hooks/useApi.tsx [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+            const isError = true;
+            toastNotify(err.response.status);
+            return isError;
```
Comment (Veryyapeee) [inline src/hooks/useApi.tsx]: You can just
`return true`

**Review by Veryyapeee — CHANGES_REQUESTED** [review]

File: src/Utils/ToastNotify.ts (hunk 1 above)
Comment (T0ffik) [reply src/Utils/ToastNotify.ts]: Done

File: src/Utils/ToastNotify.ts (hunk 2 above)
Comment (T0ffik) [reply src/Utils/ToastNotify.ts]: Done

File: src/Utils/ToastNotify.ts (hunk 3 above)
Comment (T0ffik) [reply src/Utils/ToastNotify.ts]: Done

File: src/Utils/types.ts (hunk 4 above)
Comment (T0ffik) [reply src/Utils/types.ts]: Done

File: src/hooks/useApi.tsx (hunk 5 above)
Comment (T0ffik) [reply src/hooks/useApi.tsx]: Done

File: src/hooks/useApi.tsx (hunk 6 above)
Comment (T0ffik) [reply src/hooks/useApi.tsx:19]: Done

File: src/Axios/Axios.ts (hunk 7 above)
Comment (T0ffik) [reply src/Axios/Axios.ts:6]: Done

File: src/Utils/types.ts (hunk 8 above)
Comment (T0ffik) [reply src/Utils/types.ts]: Done

File: src/hooks/useApi.tsx (hunk 9 above)
Comment (T0ffik) [reply src/hooks/useApi.tsx]: Done

File: src/hooks/useApi.tsx (hunk 10 above)
Comment (T0ffik) [reply src/hooks/useApi.tsx]: Done

**Comment by Veryyapeee** [comment]
Pipeline [*]

**Comment by T0ffik** [comment]
> Pipeline [*]

I see but locally every thing is ok ;_;

**Comment by Veryyapeee** [comment]
Also, the "endpoints" are just "endpoints", not "EndPoints" (about the folder name ~ change that pls)

File: src/hooks/useApi.tsx [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+import axios, { AxiosResponse } from 'axios';
+import { useQuery } from 'react-query';
+
+import { toastNotify } from '../utils/ToastNotify';
+
+export function useRequest<T, G>(
+    callback: (config: G) => Promise<AxiosResponse<T, G>>,
+    args: G,
+    tag: string,
+    toastToggle: boolean
```
Comment (Veryyapeee) [inline src/hooks/useApi.tsx]: that should be optional with default value of false

Also I think that the arguments could be optional too, don't we have any request which doesn't require any request body? (some GET or smth?)

**Review by Veryyapeee — CHANGES_REQUESTED** [review]

File: src/hooks/useApi.tsx (hunk 11 above)
Comment (T0ffik) [reply src/hooks/useApi.tsx]: Done

**Review by Veryyapeee — APPROVED** [review]

**Review by Mariusz-web — APPROVED** [review]

**Pull request merged**